x-pack/plugins/fleet/server/routes/package_policy/handlers.ts

```diff
@@ -82,6 +93,70 @@ export const getOnePackagePolicyHandler: RequestHandler<
   }
 };
 
+export const getOrphanedPackagePolicies: RequestHandler<undefined, undefined> = async (
+  context,
+  request,
+  response
+) => {
+  const soClient = (await context.core).savedObjects.client;
+  try {
+    const installedPackages = await getInstallations(soClient, {
+      perPage: SO_SEARCH_LIMIT,
+      filter: `
+        ${PACKAGES_SAVED_OBJECT_TYPE}.attributes.install_status:${installationStatuses.Installed}
+    `,
+    });
+    const orphanedPackagePolicies: PackagePolicy[] = [];
+    const packagePolicies = await packagePolicyService.list(soClient, {
+      perPage: SO_SEARCH_LIMIT,
+    });
+    const packagePoliciesByPackage = groupBy(packagePolicies.items, 'package.name');
+    const agentPolicies = await agentPolicyService.list(soClient, {
+      perPage: SO_SEARCH_LIMIT,
+    });
+    const agentPoliciesById = keyBy(agentPolicies.items, 'id');
+    const usedPackages = installedPackages.saved_objects.filter(
+      ({ attributes: { name } }) => !!packagePoliciesByPackage[name]
+    );
+    usedPackages.forEach(({ attributes: { name, version } }) => {
+      packagePoliciesByPackage[name].forEach((packagePolicy) => {
+        if (!agentPoliciesById[packagePolicy.policy_id]) {
+          orphanedPackagePolicies.push(packagePolicy);
+        }
+      });
+    });
+
+    return response.ok({
+      body: {
+        items: orphanedPackagePolicies,
+        total: orphanedPackagePolicies.length,
+      },
+    });
+  } catch (error) {
+    return defaultIngestErrorHandler({ error, response });
+  }
+};
+
+export const deleteOrphanedIntegrationPolicyHandler: RequestHandler<
```

Comment:
Looks like we can delete any integration policy using that endpoint not just orphaned, do you think we can use the existing delete integration policy endpoint instead? (and maybe fix it if it's not working when the agent policy is missing)

Reply:
I removed the delete orphaned endpoint in https://github.com/elastic/kibana/pull/131322/commits/2b0144d7327d09ea0da7bf85d4f775a86c0a04cd and refactored the existing delete endpoint slightly to allow us to force delete package policies w/o checking the agent policy id.